src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/DownstreamBuildAction.java

```diff
@@ -6,53 +6,80 @@
 import hudson.model.Item;
 import hudson.model.ItemGroup;
 import hudson.model.Run;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
 import org.jenkinsci.plugins.workflow.graph.FlowNode;
 import org.springframework.security.access.AccessDeniedException;
 
 /**
- * Tracks a downstream build triggered by the {@code build} step, as well as the {@link FlowNode#getId} of the step.
+ * Tracks downstream builds triggered by the {@code build} step, as well as the {@link FlowNode#getId} of the step.
  *
  * @see BuildUpstreamCause
  */
 public final class DownstreamBuildAction extends InvisibleAction {
-    private final String flowNodeId;
-    private final String jobFullName;
-    private Integer buildNumber;
+    private final Map<String, DownstreamBuild> downstreamBuilds = new LinkedHashMap<>();
```

Comment:
Works, but note that this will look a bit funny in XStream format. I would suggest using an `ArrayList` and if any indexing is necessary (unlikely, since median length is 1) do that in memory.

Reply:
https://github.com/jenkinsci/pipeline-build-step-plugin/pull/128/commits/339adf57162379e9d3ed75c78c02c7e483f7ce32